sdk/substrate/build.rs

```diff
@@ -0,0 +1,5 @@
+fn main() {
+    let output = std::process::Command::new("git").args(["rev-parse", "HEAD"]).output().unwrap();
+    let git_hash = String::from_utf8(output.stdout).unwrap();
+    println!("cargo:rustc-env=GIT_HASH={git_hash}");
+}
```

Comment:
This looks familiar, is it really necessary in multiple places?

Reply:
Yes, it is used here: https://github.com/subspace/pulsar/pull/292/files#diff-b929a80fce679ad8cae08b07576e22664294e04702e37731794ba6d1d3ebd0baR110 as default value for impl version.